JavaScript/Adhan.js

```diff
@@ -769,58 +771,56 @@
-        var month = this.getUTCMonth();
-        var day = this.getUTCDate() + days;
-        var hours = this.getUTCHours();
-        var minutes = this.getUTCMinutes();
-        var seconds = this.getUTCSeconds();
+    function dateByAddingDays(date, days) {
+        var year = date.getUTCFullYear();
+        var month = date.getUTCMonth();
+        var day = date.getUTCDate() + days;
+        var hours = date.getUTCHours();
+        var minutes = date.getUTCMinutes();
+        var seconds = date.getUTCSeconds();
         return new Date(Date.UTC(year, month, day, hours, minutes, seconds));
     }
 
-    Date.prototype.dateByAddingHours = function(hours) {
-        return this.dateByAddingMinutes(hours * 60);
+    function dateByAddingHours(date, hours) {
+        return dateByAddingMinutes(date, hours * 60);
     }
 
-    Date.prototype.dateByAddingMinutes = function(minutes) {
-        return this.dateByAddingSeconds(minutes * 60);
+    function dateByAddingMinutes(date, minutes) {
+        return dateByAddingSeconds(date, minutes * 60);
     }
 
-    Date.prototype.dateByAddingSeconds = function(seconds) {
-        return new Date(this.getTime() + (seconds * 1000));
+    function dateByAddingSeconds(date, seconds) {
+        return new Date(date.getTime() + (seconds * 1000));
     }
 
-    Date.prototype.roundedMinute = function() {
-        var seconds = this.getUTCSeconds();
+    function roundedMinute(date) {
+        var seconds = date.getUTCSeconds();
         var offset = seconds >= 30 ? 60 - seconds : -1 * seconds;
-        return this.dateByAddingSeconds(offset);
+        return dateByAddingSeconds(date, offset);
     }
 
-    Date.prototype.dayOfYear = function() {
-        var dayOfYear = 0;
-        var feb = Astronomical.isLeapYear(this.getFullYear()) ? 29 : 28;
+    function dayOfYear(date) {
+        var returnedDayOfYear = 0;
+        var feb = Astronomical.isLeapYear(date.getFullYear()) ? 29 : 28;
         var months = [31, feb, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
-        for (var i = 0; i < this.getMonth(); i++) {
-            dayOfYear += months[i];
+        for (var i = 0; i < date.getMonth(); i++) {
+            returnedDayOfYear += months[i];
         }
 
-        dayOfYear += this.getDate();
+        returnedDayOfYear += date.getDate();
 
-        return dayOfYear;
+        return returnedDayOfYear;
     }
 
-    Date.prototype.julianDate = function() {
-        return Astronomical.julianDay(this.getFullYear(), this.getMonth() + 1, this.getDate(), this.getHours() + (this.getMinutes() / 60));
+    function julianDate(date) {
+        return Astronomical.julianDay(date.getFullYear(), date.getMonth() + 1, date.getDate(), date.getHours() + (date.getMinutes() / 60));
     }
 
     //
     // Polyfill
     //
 
-    Math.trunc = Math.trunc || function(x) {
-      return x < 0 ? Math.ceil(x) : Math.floor(x);
-    }
+    var trunc = trunc || function (x) { return x < 0 ? Math.ceil(x) : Math.floor(x); };
```

Comment:
Ah, that was a mistake. I've fixed it so it uses `Math.trunc` if it is available. However, we should never **set** `Math.trunc` because it is a global side effect - the user is not expecting this library to polyfill Math.trunc.